parachain/pallets/incentivized-channel/src/outbound/mod.rs

```diff
@@ -26,24 +26,40 @@ use snowbridge_core::{types::AuxiliaryDigestItem, ChannelId};
 pub use weights::WeightInfo;
 
 /// Wire-format for committed messages
-#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug, TypeInfo)]
-pub struct MessageBundle {
+#[derive(
+	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
+)]
+#[scale_info(skip_type_params(M, N))]
+#[codec(mel_bound())]
+pub struct MessageBundle<M: Get<u32>, N: Get<u32>> {
```

Comment:
Why do we use both `skip_type_params` and `mel_bound`? My understanding of `mel_bound` is that it automatically skips all type params.

Reply:
Yeah its a bit complicated. They are used for different traits we are trying to derive. `#[scale_info(skip_type_params(M, N))]` to skip type bounds for `TypeInfo`, and `#[codec(mel_bound())]` to skip type bounds for `MaxEncodedLen`.

This answer it explains it some more: https://substrate.stackexchange.com/a/620/702